spec/attribute.dd

```diff
@@ -695,20 +695,27 @@ $(P
 $(H3 $(LNAME2 abstract, $(D abstract) Attribute))
 
 $(P
-        If a class is abstract, it cannot be instantiated
-        directly. It can only be instantiated as a base class of
-        another, non-abstract, class.
+        An abstract member function must be overridden by a derived class.
+        Only virtual member functions may be declared abstract; non-virtual 
+        member functions and free-standing functions cannot be declared 
+        abstract.
 )
+
 $(P
-        Classes become abstract if they are defined within an
-        abstract attribute, or if any of the virtual member functions
-        within it are declared as abstract.
+        Classes become abstract if any of the virtual member functions
+        within it are declared as abstract or if they are defined within an
+        abstract attribute. 
+        Note that an abstract class may also contain non-virtual member functions.
 )
+
 $(P
-        Non-virtual functions cannot be declared as abstract.
+        If a class is abstract, it cannot be instantiated
```

Comment:
I would prefer this shorter, (and more correct) sentence: `As long as a class contains abstract member functions it cannot be instantiated.`

From your sentence it seems like

```D
class Foo
{
    abstract void foo1();
    abstract void foo2();
}

class Bar: Foo
{
    override void foo1(){}
}

Bar bar = new Bar; 
```

is possible, which is obviously not true.